src/CoreBundle/DependencyInjection/Compiler/FormCompilerPass.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\CoreBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+/**
+ * @author Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * @deprecated since sonata-project/core-bundle 3.19, to be removed in 4.0.
+ */
+class FormCompilerPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container)
+    {
+        $this->registerDateAliases($container);
+        $this->registerFormTypeAlias($container);
+        $this->forceUserToMoveConfig($container);
+    }
+
+    public function registerDateAliases(ContainerBuilder $container)
+    {
+        $momentFormatConventerAlias = $container
+            ->setAlias('sonata.core.date.moment_format_converter', 'sonata.form.date.moment_format_converter')
+            ->setPublic(true)
+            ->setDeprecated(
+                true,
+                'The "%service_id%" service is deprecated since sonata-project/core-bundle 3.19 and will be removed in 4.0. Use "%alias_id%" instead.'
```

Comment:
This is what this message is producing: `The "%service_id%" service is deprecated since sonata-project/core-bundle 3.19 and will be removed in 4.0. Use "sonata.core.date.moment_format_converter" instead. It is being referenced by the "sonata.core.form.type.date_picker_legacy" service.`

I think it should be: `The "%alias_id%" service is deprecated since sonata-project/core-bundle 3.19 and will be removed in 4.0. Use "sonata.form.date.moment_format_converter" instead.`

Reference: https://travis-ci.org/github/sonata-project/SonataAdminBundle/jobs/693376005#L791

It applies for all messages

Reply:
This message are correct in xml and should be good in PHP too. I will check it when i find free time.